extensions/amp-form/0.1/amp-form.js

```diff
@@ -107,16 +127,17 @@ export class AmpForm {
       return;
     }
 
-    const shouldValidate = !this.form_.hasAttribute('novalidate');
-    if (shouldValidate &&
-        this.form_.checkValidity && !this.form_.checkValidity()) {
-      e.preventDefault();
-      // TODO(#3776): Use .mutate method when it supports passing state.
-      this.vsync_.run({
-        measure: undefined,
-        mutate: reportValidity,
-      }, {form: this.form_});
-      return;
+    if (this.shouldValidate_) {
+      checkUserValidityOnSubmission_(this.form_);
+      if (this.form_.checkValidity && !this.form_.checkValidity()) {
```

Comment:
Yes. Unfortunately with my current setup the boolean returned from `checkUserValidity` means something else (should the user-validity propagate to ancestors or not) - than what `checkValidity` returns (is the element valid or not). 

I should probably change that to keep it matched and figure a different way to propagate validity to ancestors. Will think some more about this and try to refactor this.